nifi-nar-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/poi/ConvertExcelToCSVProcessor.java

```diff
@@ -0,0 +1,409 @@
+
+                                    for (int i = 0; i < desiredSheets.length; i++) {
+                                        //If the sheetName is a desired one parse it
+                                        if (sheetName.equalsIgnoreCase(desiredSheets[i])) {
+                                            handleExcelSheet(session, flowFile, sst, sheet, sheetName);
+                                            break;
+                                        }
+                                    }
+                                }
+                            } else {
+                                getLogger().debug("Excel document was parsed but no sheets with the specified desired names were found.");
+                            }
+
+                        } else {
+                            //Get all of the sheets in the document.
+                            while (iter.hasNext()) {
+                                handleExcelSheet(session, flowFile, sst, iter.next(), iter.getSheetName());
+                            }
+                        }
+                    } catch (Exception ex) {
+                        getLogger().error(ex.getMessage());
+                    }
+                }
+            });
+
+        } catch (Exception ex) {
+            getLogger().error(ex.getMessage());
+            session.transfer(flowFile, FAILURE);
+        } finally {
+            session.transfer(flowFile, ORIGINAL);
+        }
+    }
+
+
+    /**
+     * Handles an individual Excel sheet from the entire Excel document. Each sheet will result in an individual flowfile.
+     *
+     * @param session
+     *  The NiFi ProcessSession instance for the current invocation.
+     */
+    private void handleExcelSheet(ProcessSession session, FlowFile originalParentFF,
+            SharedStringsTable sst, final InputStream sheetInputStream, String sName) {
+
+        FlowFile ff = session.create();
+        try {
+
+            XMLReader parser =
+                    XMLReaderFactory.createXMLReader(
+                            "org.apache.xerces.parsers.SAXParser"
+                    );
+            ExcelSheetRowHandler handler = new ExcelSheetRowHandler(sst);
+            parser.setContentHandler(handler);
+
+            ff = session.write(ff, new OutputStreamCallback() {
+                @Override
+                public void process(OutputStream out) throws IOException {
+                    InputSource sheetSource = new InputSource(sheetInputStream);
+                    try {
+                        ExcelSheetRowHandler eh = (ExcelSheetRowHandler) parser.getContentHandler();
+                        eh.setFlowFileOutputStream(out);
+                        parser.setContentHandler(eh);
+                        parser.parse(sheetSource);
+                        sheetInputStream.close();
+                    } catch (Exception ex) {
+                        getLogger().error(ex.getMessage());
```

Comment:
Please include a comment and the full exception trace, like `getLogger().error("Attempt to do X failed", ex)`.